backend/app/routers/listeners.py

```diff
@@ -374,10 +391,66 @@ async def edit_listener(
     raise HTTPException(status_code=404, detail=f"listener {listener_id} not found")
 
 
+@router.put("/{listener_id}/enable", response_model=EventListenerOut)
+async def enable_listener(
+    listener_id: str,
+    user_id=Depends(get_user),
+    allow: bool = Depends(ListenerAuthorization()),
+):
+    """Enable an Event Listener. Only admins can enable listeners.
+
+    Arguments:
+        listener_id -- UUID of the listener to be enabled
+    """
+    return await set_active_flag(listener_id, True)
+
+
+@router.put("/{listener_id}/disable", response_model=EventListenerOut)
+async def disable_listener(
+    listener_id: str,
+    user_id=Depends(get_user),
+    allow: bool = Depends(ListenerAuthorization()),
+):
+    """Enable an Event Listener. Only admins can enable listeners.
+
+    Arguments:
+        listener_id -- UUID of the listener to be enabled
+    """
+    return await set_active_flag(listener_id, False)
+
+
+@router.put("/{listener_id}/toggle", response_model=EventListenerOut)
```

Comment:
if this toggles, do we need the other two methods above?

Reply:
I am using this from frontend to toggle the flag. I was thinking if there is a use-case where somebody explicitly calls a backend endpoint to enable/disable the listener irrespective of the active flag. Hence, I created the separate endpoints, but all use the same function internally.

Reply:
<img width="1705" alt="image" src="https://github.com/clowder-framework/clowder2/assets/13950475/44be8ecb-db90-4671-869a-386b586ab434">

After I select the extractors, what do I do to enable/disable? I don't see any button around. I should already be in admin mode.

Reply:
@longshuicy Clicking on the checkboxes does the job. In this case, both the extractors are enabled.